it/src/main/java/org/corfudb/universe/node/server/docker/DockerCorfuServer.java

```diff
@@ -0,0 +1,403 @@
+        String[] ports = {String.valueOf(params.getPort())};
+        Map<String, List<PortBinding>> portBindings = new HashMap<>();
+        for (String port : ports) {
+            List<PortBinding> hostPorts = new ArrayList<>();
+            hostPorts.add(PortBinding.of(ALL_NETWORK_INTERFACES, port));
+            portBindings.put(port, hostPorts);
+        }
+
+        HostConfig hostConfig = HostConfig.builder()
+                .privileged(true)
+                .portBindings(portBindings)
+                .build();
+
+        // Compose command line for starting Corfu
+        String cmdLine = String.format(
+                "java -cp *.jar org.corfudb.infrastructure.CorfuServer %s",
+                getCommandLineParams()
+        );
+
+        return ContainerConfig.builder()
+                .hostConfig(hostConfig)
+                .image(IMAGE_NAME)
+                .hostname(params.getName())
+                .exposedPorts(ports)
+                .cmd("sh", "-c", cmdLine)
+                .build();
+    }
+
+    private void addShutdownHook() {
+        // Just in case a test failed and didn't kill the container
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            try {
+                destroy();
+            } catch (Exception e) {
+                log.debug("Corfu server shutdown hook. Can't kill container: {}", params.getName());
+            }
+        }));
+    }
+
+    /**
+     * Run `docker exec` on a container
+     */
+    private void execCommand(String... command) throws DockerException, InterruptedException {
+        log.info("Executing docker command: {}", String.join(" ", command));
+
+        docker.execCreate(
+                params.getName(),
+                command,
+                ExecCreateParam.attachStdout(),
+                ExecCreateParam.attachStderr()
+        );
+    }
+
+    /**
+     * Collect logs from container and write to the log directory
+     */
+    private void collectLogs() {
+        if (!loggingParams.isEnabled()) {
+            log.debug("Logging is disabled");
+            return;
+        }
+
+        File serverLogDir = loggingParams.getServerLogDir().toFile();
+        if (!serverLogDir.exists() && serverLogDir.mkdirs()) {
+            log.info("Created new corfu log directory at {}.", serverLogDir);
+        }
+
+        log.debug("Collect logs for: {}", params.getName());
+
+        try (LogStream stream = docker.logs(params.getName(), LogsParam.stdout(), LogsParam.stderr())) {
+            String logs = stream.readFully();
+
+            if (StringUtils.isEmpty(logs)) {
+                log.warn("Empty logs from container: {}", params.getName());
+            }
+
+            Path filePathObj = loggingParams.getServerLogDir().resolve(params.getName() + ".log");
+            Files.write(filePathObj, logs.getBytes(), StandardOpenOption.CREATE, StandardOpenOption.APPEND);
+        } catch (InterruptedException | DockerException | IOException e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either re-interrupt this method or rethrow the "InterruptedException". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2142)